packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1708,7 +1709,7 @@ class TabPageSelector extends StatelessWidget {
       animation: animation,
       builder: (BuildContext context, Widget? child) {
         return Semantics(
-          label: 'Page ${tabController.index + 1} of ${tabController.length}',
+          label: localizations.tabLabel(tabIndex: tabController.index + 1, tabCount: tabController.length),
```

Comment:
This changes the semantics label from `Page` to `Tab`

Reply:
update, I reached out to the original developer that adds this line, the label should be tab instead of page, so we should be fine